tutorials/Inference-Bert-Model-for-High-Performance-with-ONNX-Runtime-on-AzureML.ipynb

```diff
@@ -0,0 +1,690 @@
+   "metadata": {},
+   "source": [
+    "Copyright (c) Microsoft Corporation. All rights reserved.  \n",
+    "Licensed under the MIT License.\n",
+    "\n",
+    "# Inference Bert Model for High Performance with ONNX Runtime on AzureML #\n",
+    "\n",
+    "This tutorial includes how to pretrain and finetune Bert models using AzureML, convert it to ONNX, and then deploy the ONNX model with ONNX Runtime through Azure ML. \n",
+    "\n",
+    "## Roadmap\n",
+    "\n",
+    "0. **Prerequisites** to set up your Azure ML work environments.\n",
+    "1. **Pre-train and finetune Bert model** from other framework using Azure ML.\n",
+    "2. **Convert the Bert model to ONNX**\n",
+    "3. **Deploy Bert model using ONNX Runtime and AzureML**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 0 - Prerequisites\n",
+    "If you are using an [Azure Machine Learning Notebook VM](https://docs.microsoft.com/en-us/azure/machine-learning/service/quickstart-run-cloud-notebook), you are all set. Otherwise, refer to the [configuration Notebook](https://github.com/Azure/MachineLearningNotebooks/blob/56e0ebc5acb9614fac51d8b98ede5acee8003820/configuration.ipynb) first if you haven't already to establish your connection to the AzureML Workspace. Prerequisites are:\n",
+    "* Azure subscription\n",
+    "* Azure Machine Learning Workspace\n",
+    "* Azure Machine Learning SDK\n",
+    "\n",
+    "Also to make the best use of your time, make sure you have done the following:\n",
+    "* Understand the [architecture and terms](https://docs.microsoft.com/azure/machine-learning/service/concept-azure-machine-learning-architecture) introduced by Azure Machine Learning\n",
+    "* [Azure Portal](https://portal.azure.com) allows you to track the status of your deployments."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1 - Pretrain and Finetune a Bert Model (PyTorch)\n",
+    "\n",
+    "If you'd like to pre-train and finetune a Bert model from scratch, follow the instructions in [\n",
+    "Pretraining of the BERT model](https://github.com/microsoft/AzureML-BERT/blob/master/pretrain/PyTorch/notebooks/BERT_Pretrain.ipynb) to pretrain a Bert model in PyTorch using AzureML. Once you have the pretrained model, refer to [AzureML Bert Eval Squad](https://github.com/microsoft/AzureML-BERT/blob/master/finetune/PyTorch/notebooks/BERT_Eval_SQUAD.ipynb) or [AzureML Bert Eval GLUE](https://github.com/microsoft/AzureML-BERT/blob/master/finetune/PyTorch/notebooks/BERT_Eval_GLUE.ipynb) to finetune your model with your desired dataset. Save the output model to Azure DataStore Blob."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2 - Export the model to ONNX (PyTorch)\n",
+    "\n",
+    "Once you have your model ready, download it to local directory from Azure DataStore Blob. The following tutorial will use `bert-base-squad` as the example.\n",
```

Comment:
why do you need to download it? BERT models are rather large so it would be better to do the conversion in the cloud where the model already is, especially since it's going to be deployed to the cloud anyways.

if you need them to download, then you also need to explain what steps to take to setup the local environment

Reply:
If the model is in a run already, it will be much quicker to use run.register_model